open-feature/flagd
# feat: expose Impression metric

## This PR


- Introduces Impressions split by Flag Key and Variant as a Prometheus Metric exposed via OpenTelemetry

## Related Issues
Fixes #501

## How to test
Run flagd via `make run`
Run 
```bash
curl -X POST "localhost:8013/schema.v1.Service/ResolveString" -d '{"flagKey":"fibAlgo","context":{}}' -H "Content-Type: application/json"
curl -X POST "localhost:8013/schema.v1.Service/ResolveString" -d '{"flagKey":"myStringFlag","context":{}}' -H "Content-Type: application/json"
```
Open <http://localhost:8014/metrics> and see something like the following
```
# HELP impressions_total The number of evaluation for a given flag
# TYPE impressions_total counter
impressions_total{feature_flag_key="fibAlgo",feature_flag_provider_name="flagd",feature_flag_variant="recursive",otel_scope_name="openfeature/flagd",otel_scope_version=""} 3
impressions_total{feature_flag_key="myString",feature_flag_provider_name="flagd",feature_flag_variant="",otel_scope_name="openfeature/flagd",otel_scope_version=""} 1
impressions_total{feature_flag_key="myStringFlag",feature_flag_provider_name="flagd",feature_flag_variant="key1",otel_scope_name="openfeature/flagd",otel_scope_version=""} 3
```
